core/primitives/src/types.rs

```diff
@@ -990,3 +991,23 @@ pub enum TrieCacheMode {
     /// and thus users (potentially different) are not required to pay twice for retrieval of the same node.
     CachingChunk,
 }
+
+/// Counts accessed trie nodes during tx/receipt execution for proper storage costs charging.
+#[derive(Debug)]
+pub struct TrieNodesCount {
+    /// Number of nodes read from storage or shard cache.
+    pub touches: u64,
+    /// Number of nodes read from the chunk cache.
+    pub chunk_cache_reads: u64,
```

Comment:
It seems that the names could be better:

* they seem to describe non-essential properties
* the two names should be symmetric, but they aren't

I'd maybe suggest something like

```
/// Cheap trie node reads which are guaranteed to be served from RAM.
pub mem_reads: u64,
/// Potentially expensive trie node reads which are served from disk in the worst case.
pub db_reads: u64,
```

Reply:
Though, even if we do renaming along the lines, it *might* makes sense to do a separate PR for that and take a holistic look at the naming. It seems that the core problem is that we use chunk/block terminology inside the trie, but the trie exists on a level of abstraction *below* chunks and blocks.